src/python/pants/jvm/dependency_inference/artifact_mapper.py

```diff
@@ -243,10 +244,41 @@ class AllJvmTypeProvidingTargets(Targets):
     pass
 
 
+class AllJvmArtifactPackagesTargets(Targets):
+    pass
+
+
+class JvmArtifactPackagesMapping(FrozenDict[UnversionedCoordinate, FrozenOrderedSet[str]]):
+    @classmethod
+    def from_dict(
+        cls, data: Mapping[UnversionedCoordinate, OrderedSet[str]]
+    ) -> JvmArtifactPackagesMapping:
+        return JvmArtifactPackagesMapping(
+            {key: FrozenOrderedSet(value) for key, value in data.items()}
+        )
+
+
 @rule(desc="Find all jvm_artifact targets in project", level=LogLevel.DEBUG)
 def find_all_jvm_artifact_targets(targets: AllTargets) -> AllJvmArtifactTargets:
     return AllJvmArtifactTargets(
-        tgt for tgt in targets if tgt.has_fields((JvmArtifactGroupField, JvmArtifactArtifactField))
+        tgt
+        for tgt in targets
+        if tgt.has_fields(
+            (JvmArtifactGroupField, JvmArtifactArtifactField, JvmArtifactVersionField)
```

Comment:
added `JvmArtifactVersionField` here to make sure it won't select `jvm_artifact_packages`.